src/internal/operators/windowCount.ts

```diff
@@ -69,79 +70,66 @@ import { lift } from '../util/lift';
  * @name windowCount
  */
 export function windowCount<T>(windowSize: number, startWindowEvery: number = 0): OperatorFunction<T, Observable<T>> {
-  return function windowCountOperatorFunction(source: Observable<T>) {
-    return lift(source, new WindowCountOperator<T>(windowSize, startWindowEvery));
-  };
-}
-
-class WindowCountOperator<T> implements Operator<T, Observable<T>> {
-  constructor(private windowSize: number, private startWindowEvery: number) {}
-
-  call(subscriber: Subscriber<Observable<T>>, source: any): any {
-    return source.subscribe(new WindowCountSubscriber(subscriber, this.windowSize, this.startWindowEvery));
-  }
-}
-
-/**
- * We need this JSDoc comment for affecting ESDoc.
- * @ignore
- * @extends {Ignored}
- */
-class WindowCountSubscriber<T> extends Subscriber<T> {
-  private windows: Subject<T>[] = [new Subject<T>()];
-  private count: number = 0;
-
-  constructor(protected destination: Subscriber<Observable<T>>, private windowSize: number, private startWindowEvery: number) {
-    super(destination);
-    destination.next(this.windows[0]);
-  }
+  const startEvery = startWindowEvery > 0 ? startWindowEvery : windowSize;
 
-  protected _next(value: T) {
-    const startWindowEvery = this.startWindowEvery > 0 ? this.startWindowEvery : this.windowSize;
-    const destination = this.destination;
-    const windowSize = this.windowSize;
-    const windows = this.windows;
-    const len = windows.length;
+  return (source: Observable<T>) =>
+    lift(source, function (this: Subscriber<Observable<T>>, source: Observable<T>) {
+      const subscriber = this;
+      let windows = [new Subject<T>()];
+      let starts: number[] = [];
+      let count = 0;
 
-    for (let i = 0; i < len && !this.closed; i++) {
-      windows[i].next(value);
-    }
-    const c = this.count - windowSize + 1;
-    if (c >= 0 && c % startWindowEvery === 0 && !this.closed) {
-      windows.shift()!.complete();
-    }
-    if (++this.count % startWindowEvery === 0 && !this.closed) {
-      const window = new Subject<T>();
-      windows.push(window);
-      destination.next(window);
-    }
-  }
+      // Open the first window.
+      subscriber.next(windows[0].asObservable());
 
-  protected _error(err: any) {
-    const windows = this.windows;
-    if (windows) {
-      while (windows.length > 0 && !this.closed) {
-        windows.shift()!.error(err);
-      }
-    }
-    this.destination.error(err);
-  }
+      const outerSubscriber = new OperatorSubscriber(
```

Comment:
nitpick: This variable is used in only one place: the subscribe call. Other operator implementations pass the `new OperatorSubscriber` directly to `subscribe`.